xformers/factory/__init__.py

```diff
@@ -1,10 +1,20 @@
+# register factory configs
+from hydra.core.config_store import ConfigStore
+
 from xformers.components import MultiHeadDispatchConfig  # noqa
 from xformers.components.attention import AttentionConfig  # noqa
 from xformers.components.feedforward import FeedforwardConfig  # noqa
 from xformers.components.positional_embedding import PositionEmbeddingConfig  # noqa
 
-from .block_factory import xFormerDecoderBlock  # noqa
-from .block_factory import xFormerDecoderConfig  # noqa
-from .block_factory import xFormerEncoderBlock  # noqa
+from .block_factory import xFormerDecoderConfig  # noqa; noqa
+from .block_factory import xFormerEncoderBlock, xFormerDecoderBlock  # noqa
 from .block_factory import xFormerEncoderConfig  # noqa
-from .model_factory import xFormer, xFormerConfig  # noqa
+from .block_factory import xFormerBlockConfig
+from .model_factory import xFormer, xFormerConfig, xFormerStackConfig  # noqa
+
+cs = ConfigStore.instance()
```

Comment:
no super strong opinion here, but I'm wondering whether this could exist on the side (in a dedicated hydra part, so that it's not forced on people who would like pure pytorch code) ?

Reply:
makes sense! ive updated the approach in #93 :)